include/igl/arap_linear_block.h

```diff
@@ -43,32 +43,32 @@ namespace igl
   //   Kd  #V by #V/#F block of the linear constructor matrix corresponding to
   //     coordinate d
   //
-  template <typename MatV, typename MatF, typename Scalar>
+  template <typename MatV, typename MatF, typename MatK>
   IGL_INLINE void arap_linear_block(
     const MatV & V,
     const MatF & F,
     const int d,
     const igl::ARAPEnergyType energy,
-    Eigen::SparseMatrix<Scalar> & Kd);
+    MatK & Kd);
```

Comment:
Can these be either `SparseMatrixBase` or `SparseCompressedBase`?

Reply:
Note: I think we'll need to bump the version of Eigen for this.

Reply:
doesnt work. `SparseCompressedBase` does not support resize reserve etc